yarn-project/aztec.js/src/deployment/protocol_contracts.ts

```diff
@@ -1,17 +1,19 @@
-import { getCanonicalClassRegisterer } from '@aztec/protocol-contracts/class-registerer';
-import { getCanonicalInstanceDeployer } from '@aztec/protocol-contracts/instance-deployer';
+import { ProtocolContractAddress } from '@aztec/protocol-contracts';
 
 import { UnsafeContract } from '../contract/unsafe_contract.js';
 import type { Wallet } from '../wallet/index.js';
 
 /** Returns a Contract wrapper for the class registerer. */
 export async function getRegistererContract(wallet: Wallet) {
-  const { artifact, instance } = await getCanonicalClassRegisterer();
-  return new UnsafeContract(instance, artifact, wallet);
+  const { contractInstance } = await wallet.getContractMetadata(ProtocolContractAddress.ContractClassRegisterer);
```

Comment:
@spalladino I need to decouple protocol-contracts from `aztec.js`,  due to the pxe now receiving the artifact provider (so the user is in control of how they're loaded), I think this is reasonable, this code would eventually fail if the canonical contracts were not registered. WDYT?

Reply:
I think that's fine. It puts an extra burden on the wallet though, but it's not too bad. And I like it makes aztec.js more lightweight!